library/kani/src/vec.rs

```diff
@@ -0,0 +1,12 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+use crate::{any, Invariant};
+
+/// Generates an arbitrary vector given a length.
+pub fn any_vec<const VEC_LENGTH: usize, T>() -> Vec<T>
```

Comment:
Not sure I understand the `const` comment, but I do like the idea of using maximum size, like any_slice does.

Reply:
I guess the question is do we want vec of exactly that length, or nondet up to a max length

Reply:
> But we don't seem to provide any way to do that via slice do we, and that (I assume) is why you wound up with the const...

Yea, not without modifying `slice.rs` so I can see `AnySlice`'s internals. I try to see if I can implement  `From<AnySlice<T, MAX_SLICE_LENGTH>> for Vec<T>`, which converts `AnySlice` to `*[T]`, then `Box<[T]>` then `Vec<T>`. `any_vec` will just call that `From::from`.

Reply:
After some investigation, 2/4 test cases are working, 1 is still not working (#1322) and 1 migrated to the newly forked `exact_vec` API.

After talking with @danielsn, we will expose `exact_vec` that generates exact-length vectors since the performance is significantly better than `any_vec + assume`. See #1322 for details